src/Type/Constant/ConstantArrayType.php

```diff
@@ -730,6 +730,25 @@ public function reverse(bool $preserveKeys = false): self
 		return $preserveKeys ? $reversed : $reversed->reindex();
 	}
 
+	/** @param positive-int $length */
+	public function chunk(int $length, bool $preserveKeys = false): self
+	{
+		$builder = ConstantArrayTypeBuilder::createEmpty();
+
+		$keyTypesCount = count($this->keyTypes);
+		for ($i = 0; $i < $keyTypesCount; $i += $length) {
+			$chunk = $this->slice($i, $length, true);
+			$builder->setOffsetValueType(null, $preserveKeys ? $chunk : $chunk->getValuesArray());
+		}
+
+		$chunks = $builder->getArray();
+		if (!$chunks instanceof self) {
+			throw new ShouldNotHappenException();
```

Comment:
This can fail for some big arrays I guess so it should just default to return some general ArrayType.

Reply:
I thought that both the resulting chunks and the list containing them will never be bigger than the length of the current constant array. And therefore it should still be a constant array? Did I miss something?

Reply:
Alright, I can't think of ways of breaking it. One more thing - can you add some tests for chunking of general arrays? Especially non-empty-array - chunking it should lead to non-empty-array too?

Reply:
I also thought more about it. Currently it should never break, but e.g. if in the future a depth limit is added, then it could break because chunking increases depth by one basically.

Good idea about non-empty-array, this can get lost currently.. Will fix it in a bit